### PR TITLE
added Piece fixes #1

### LINKED ISSUE #1: Class Piece

### PR DESCRIPTION
**Pull request by SimonMattei**

### CODE REVIEW

File: src/main/java/Square.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+public class Square {
+    String name;
```
Comment (BPotet) [inline src/main/java/Square.java]: String should be private

File: src/test/java/PieceTest.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        Piece p = new Piece("Test");
+        assertEquals("Go",p.getLocation().getName());
+    }
+
+    @Test
+    void newLocationShouldBeCorrect() {
+        Piece p = new Piece("Go");
+        p.setLocation(new Square("Square 1"));
+        assertEquals("Square 1",p.getLocation().getName());
+    }
```
Comment (BPotet) [inline src/test/java/PieceTest.java:27]: - Adding test for privacy for the class
- Adding test for empty constructors or null constructors

File: src/main/java/Piece.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+public class Piece {
+    private String name;
+    private Square location;
+
+    Piece(String name){
+        this.name = name;
+        location = new Square("Go");
```
Comment (BPotet) [inline src/main/java/Piece.java:13]: Pass the "Go" in constant (like static final ...)

File: src/main/java/Piece.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    public String getName() {
+        return name;
+    }
+
+    public Square getLocation() {
+        return location;
+    }
+
+    public void setLocation(Square location) {
+        this.location = location;
```
Comment (BPotet) [inline src/main/java/Piece.java:25]: if location = null then ?

File: src/main/java/Piece.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+public class Piece {
+    private String name;
+    private Square location;
+
+    Piece(String name){
+        this.name = name;
```
Comment (BPotet) [inline src/main/java/Piece.java:12]: name could be empty or null ?

File: src/main/java/Square.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+    Square(String name){
+        this.name = name;
```
Comment (BPotet) [inline src/main/java/Square.java:5]: name could be empty or null

**Review by BPotet — CHANGES_REQUESTED** [review]
missing some checks in constructors

**Review by BPotet — APPROVED** [review]
well played Ikewolf :)

**Review by BPotet — APPROVED** [review]

**Review by BPotet — APPROVED** [review]

**Review by BPotet — APPROVED** [review]
